### PR TITLE
Pipewire 0.3.83

### PR DESCRIPTION
**Pull request by K900**
## Description of changes

Routine update, but big cleanup. I'm open to being convinced to drop the cleanup.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by K900** [comment]
@ofborg eval

**Comment by K900** [comment]
Also run-tested on my laptop with no issues.

File: pkgs/development/libraries/pipewire/default.nix [hunk 1]
```diff
@@ -170,9 +158,7 @@ let
       "-Dudevrulesdir=lib/udev/rules.d"
       "-Dinstalled_tests=enabled"
       "-Dinstalled_test_prefix=${placeholder "installedTests"}"
-      "-Dpipewire_pulse_prefix=${placeholder "pulse"}"
       "-Dlibjack-path=${placeholder "jack"}/lib"
-      "-Dlibv4l2-path=${placeholder "out"}/lib"
```
Comment (jansol) [inline pkgs/development/libraries/pipewire/default.nix:175]: Did you verify that v4l2 still works the way it is intended?

File: pkgs/development/libraries/pipewire/default.nix [hunk 2]
```diff
@@ -181,7 +167,6 @@ let
       "-Dgstreamer=${mesonEnableFeature gstreamerSupport}"
       "-Dsystemd-system-service=${mesonEnableFeature enableSystemd}"
       "-Dudev=${mesonEnableFeature (!enableSystemd)}"
-      "-Dudevrulesdir=${placeholder "out"}/lib/udev/rules.d"
```
Comment (jansol) [inline pkgs/development/libraries/pipewire/default.nix:184]: Ditto here, if you have a link to upstream meson.build changes that make this line unnecessary I'd appreciate seeing it.

**Review by jansol — APPROVED** [review]
Haven't had time to test this but I appreciate the cleanup ahead of the 1.0 release and at a glance the changes look sensible.

File: pkgs/development/libraries/pipewire/default.nix (hunk 1 above)
Comment (K900) [reply pkgs/development/libraries/pipewire/default.nix:175]: Yes, it just ends up in $out/lib now which is fine: https://gitlab.freedesktop.org/pipewire/pipewire/-/blob/master/pipewire-v4l2/src/meson.build?ref_type=heads#L18

File: pkgs/development/libraries/pipewire/default.nix (hunk 2 above)
Comment (K900) [reply pkgs/development/libraries/pipewire/default.nix:184]: That is still necessary, I think it got lost in the rebases somewhere. I'll put it back when I'm back home in like an hour.

Comment (K900) [reply pkgs/development/libraries/pipewire/default.nix:184]: Wait no never mind this is fine. It was just a duplicated line: https://github.com/NixOS/nixpkgs/pull/262148/files#diff-fdd948de77300ac589c79c5b6dd5f79e757362ad88beac84d4352da1aa7bfa12R158

Comment (jansol) [reply pkgs/development/libraries/pipewire/default.nix:184]: That duplicate is missing the "out" placeholder though. Not sure if that is needed any more.

Comment (K900) [reply pkgs/development/libraries/pipewire/default.nix:184]: It isn't, Meson will treat relative paths as relative to the prefix (which is `$out`).

**Comment by K900** [comment]
I double-checked and all the files are in the right places. Let's merge.

**Pull request merged**